doc/tutorials/acrn-secure-boot-with-grub.rst

```diff
@@ -0,0 +1,179 @@
+  a typical GRUB configuration fragment might look like this (add them to ``grub.init.cfg`` file)::
+
+	set superusers="root"
+	export superusers
+	password_pbkdf2 root GRUB_PASSWORD_HASH
+
+  Replace the ``GRUB_PASSWORD_HASH`` with the result of ``grub-mkpasswd-pbkdf2`` with your custom passphrase.
+
+  Sign grub.init.cfg file with GPG key, as result ``grub.init.cfg.sig`` file will be created::
+  gpg --homedir keys --detach-sign grub.init.cfg
+
+
+Creating Single GRUB EFI Binary
+===============================
+
+  Creating single GRUB EFI binary with buit-in modules and signed grub.init.cfg by ``grub-mkstandalone`` tool,
+  the ``--pubkey`` option adds a GPG public key which will be used for verification, and public key ``boot.key``
+  is not required anymore after ``grub-mkstandalone``, hence it can be removed.
+
+  Script Example::
+
+	#!/bin/bash
+	#
+
+	TARGET_EFI='/boot/efi/EFI/ubuntu/grubx64.efi'
+
+	# GRUB doesn't allow loading new modules from disk when secure boot is in
+	# effect, therefore pre-load the required modules.
+	MODULES="all_video archelp boot bufio configfile crypto echo efi_gop efi_uga ext2 extcmd  \
+	fat font fshelp gcry_dsa gcry_rsa gcry_sha1 gcry_sha512 gettext gfxterm linux linuxefi ls \
+	memdisk minicmd mmap mpi normal part_gpt part_msdos password_pbkdf2 pbkdf2 reboot relocator \
+	search search_fs_file search_fs_uuid search_label sleep tar terminal verify video_fb"
+
+	grub-mkstandalone \
+		--directory /usr/lib/grub/x86_64-efi \
+		--format x86_64-efi \
+		--modules "$MODULES" \
+		--pubkey ./boot.key \
+		--output ./grubx64.efi  \
+		"boot/grub/grub.cfg=./grub.init.cfg" \
+		"boot/grub/grub.cfg.sig=./grub.init.cfg.sig"
+
+	echo "writing signed grub.efi to '$TARGET_EFI'"
+	sudo cp ./grubx64.efi "$TARGET_EFI"
+
+
+..note::
+	You should backup current GRUB image in '$TARGET_EFI', then you can restore your grub in case you got stuck after updating it.
+
+
+Prepare grub.cfg
+================
+  Define the menuenty for your system in a new GRUB configration file ``grub.cfg``.
+  Example::
+
+	# @/boot/efi/grub.cfg for grub secure boot
+	 set timeout_style=menu
+	 set timeout=5
+	 set gfxmode=auto
+	 set gfxpayload=keep
+	 terminal_output gfxterm
+
+	 menuentry "ACRN Multiboot Ubuntu Service VM" --users "" --id ubuntu-service-vm {
+
+	 search --no-floppy --fs-uuid --set 3df12ea1-ef12-426b-be98-774665c7483a
+
+	 echo 'loading ACRN...'
+	 multiboot2 /boot/acrn/acrn.bin  root=PARTUUID="c8ee7d92-8935-4e86-9e12-05dbeb412ad6"
+	 module2 /boot/bzImage Linux_bzImage
+	}
+
+  Update the UUID (``--set``) and PARTUUID (``root=`` parameter) of the root partition (e.g. /dev/nvme0n1p2) according to
+  your your hardware(e.g. use ``blkid`` tool).
+
+  Copy :file:`grub.cfg` to your ESP (e.g. `/boot/efi/EFI/`).
+
+
+Sign grub.cfg and ACRN Binaries
+===============================
+
+  The :file:`grub.cfg` file and all ACRN binaries that will be loaded by GRUB **must** be signed with GPG key.
+
+  Sign script example::
```

Comment:
Better call that a "sign sequence example". It cannot really be copy-pasted into a file and ran as a script